File: java_spring/exos/exo3/src/main/java/com/example/exo3/models/ContactDTO.java
package com.example.exo3.models;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Past;
import jakarta.validation.constraints.Pattern;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.util.UUID;

@Data
@Builder
public class ContactDTO {

    private UUID id;

    @NotNull(message = "First name must not be null")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "Names must contain only letters")
    private String firstName;

    @NotNull(message = "Last name must not be null")
    @Pattern(regexp = "^[a-zA-Z]+$", message = "Names must contain only letters")
    private String lastName;

    @NotNull(message = "Email must not be null")
    @Email(message = "Invalid email")
    private String email;

    @NotNull(message = "Phone must not be null")
    @Pattern(regexp = "^[+]?\\d+$", message = "Invalid phone pattern")
    private String phone;

    @NotNull(message = "Birth date must not be null")
    @Past(message = "Birth date must be a past date")
    private LocalDate birthDate;

    private Integer age;
}

File: js_typescript/exos/tp/front/src/assets/ts/ContactDTO.ts
class Contact {
    public firstName : string;
    public lastName : string;
    public birthDate : string;
    public email : string;
    public phone : string;

    constructor(
        firstName : string,
        lastName : string,
        birthDate : string,
        email : string,
        phone : string
    ) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.email = email;
        this.phone = phone;
    }
}

export default Contact;